web/app.py

```diff
@@ -87,14 +90,29 @@ def resolve():
             'message': 'url parameter is required',
         }}, 400
 
+    canonical_url = None
     response = requests.get(url, headers=HEADERS)
+
+    # TODO: what is our strategy regarding domain redirects and selection of
+    # scrapers?  Should it use the request and/or response domain(s)?
```

Comment:
The implementation seems to have the correct approach: we select a scraper based on the domain of the response, since that will correspond to the format of the response body.

It could be argued that it is slightly confusing that we store a `url`, `domain`, and `resolves_to` -- since it may be unclear to readers whether the domain refers to the original or resolved URL.  But I think it is reasonably intuitive that the `domain` is determined prior to resolution.  We could add a comment to that effect on the `domain` field.